shelley/design-spec/delegation_design_spec.tex

```diff
@@ -852,106 +804,83 @@ \subsubsection{Reward Account Address}
-as normal addresses with no stake rights, but has different motivating
-use cases. For many scripts it is hard to determine to whom the funds
-actually belong and in such cases the simplest solution is to use an
-address with no stake rights whatsoever. For this reason we do not call
-such addresses enterprise addresses.
-
 \subsubsection{HD Wallet Structure in Shelley}
 \label{hd-wallet-structure-in-shelley}
 
-All the Shelley address formats (other than script addresses) support
-hierarchical deterministic wallets, as per BIP-32 \citep{bip32}.
+Shelley addresses with a verification key hash as payment credential
+support hierarchical deterministic wallets, as per BIP-32~\citep{bip32}.
 
-In particular this kind of wallet scheme allows implementations that can
+For multi-sig value addresses, we can use a slight generalisation of
+BIP-45~\citep{bip45}\todo{Lay out the necessary generalisation of BIP-45}.
+
+In particular, this kind of wallet scheme allows implementations that can
 do wallet restoration from seed in time that is better than linear in
 the total number of addresses in the blockchain. For details, see
 \cref{wallet-recovery-process}.
 
-\subsection{Address Recognition}
+\subsubsection{Address Recognition}
 \label{address-recognition-1}
 
-Wallets will recognise addresses that belong to them just as they would
-without delegation, by looking only at the \(\hash{vkp}\) part
-of the address.
-
-After a wallet recognises an address for which it controls the payment
-key, it will check whether the staking object \(\beta\) is set according
+Wallets will recognise addresses (other than reward addresses) that belong to
+them just as they would without delegation, by looking only at the
+payment credential (see \cref{wallet-recovery-process} for how to find
+addresses efficiently).
+
+Finding the stake credentials of a wallet (and in particular the corresponding
+reward accounts) can be done either by just reading off all the stake credentials
+of addresses found via their payment credentials, or by performing a second search,
+this time over all the registered stake addresses\footnote{As explained in
+  \cref{stake-address-registration-certificates}, stake addresses need to be
+  registered on-chain.}. The former option is quicker and easier to implement.
+However, it is conceivable to construct addresses where the payment credential
+belongs to one wallet, but the stake credential belongs to another. The stake
+credential of such addresses would only be found by the wallet that controls it via
+a dedicated search on the registered stake addresses.
+
+Once a wallet recognises an address via its payment credential, it will read its
+stake credential, and check whether it is set according
 to the current delegation preference of the wallet. If there is a
 discrepancy, it will alert the user, and ask them whether they want to
 re-delegate according to their current delegation preferences.
```

Comment:
This appears to be describing essential requirements for any wallet, rather than what Daedalus does?

If so use the word "must", and highlight somewhere (e.g. as a subsection or concordance) what are the minimal functional requirements for a valid wallet.